src/utilities/findDescriptionChanges.js

```diff
@@ -0,0 +1,142 @@
+/* eslint-disable no-restricted-syntax */
+// @flow
+
+import {
+  GraphQLInterfaceType,
+  GraphQLObjectType,
+  GraphQLEnumType,
+  GraphQLInputObjectType,
+} from '../type/definition';
+import type { GraphQLFieldMap } from '../type/definition';
+import { GraphQLSchema } from '../type/schema';
+
+/**
+ * Given two schemas, returns an Array containing descriptions of any
+ * descriptions that are new or changed and need review.
+ */
+export function findDescriptionChanges(
+  oldSchema: GraphQLSchema,
+  newSchema: GraphQLSchema,
+): Array<string> {
+  const oldTypeMap = oldSchema.getTypeMap();
+  const newTypeMap = newSchema.getTypeMap();
+
+  const descriptionChanges: Array<string> = [];
+
+  Object.keys(newTypeMap).forEach(typeName => {
+    const oldType = oldTypeMap[typeName];
+    const newType = newTypeMap[typeName];
+
+    if (newType.description) {
+      if (!oldType) {
+        descriptionChanges.push(
+          `Description added on new type ${newType.name}.`,
+        );
+      } else if (!oldType.description) {
+        descriptionChanges.push(`Description added on type ${newType.name}.`);
+      } else if (oldType.description !== newType.description) {
+        descriptionChanges.push(`Description changed on type ${newType.name}.`);
+      }
+    }
+
+    if (oldType && !(newType instanceof oldType.constructor)) {
```

Comment:
Flow cannot see this as a cross-type refinement, plus it's a bit challenging to understand. I would instead make instanceof invariants for the `oldType` within each case below

Reply:
I did it this way because spelling out the `instanceof`s actually becomes really messy in its own right to handle the cases where `oldType` is null/undefined, and where `oldType` is present but a different kind from `newType`. I'll play with it a bit to see if I can make it nicer, but any suggestions would be very welcome.

Reply:
In this case, below line 50, you could just mirror the if statement above in an invariant while including the falsey check:

```
invariant(
  !oldType ||
  oldType instanceof GraphQLObjectType ||
  oldType instanceof GraphQLInterfaceType ||
  oldType instanceof GraphQLInputObjectType,
  'Expected oldType to also have fields'
);
```